src/Redux/State/History.js

```diff
@@ -68,4 +88,18 @@ export default createReducer(INITIAL_STATE, {
     ],
     totalPages,
   }),
+
+  [HISTORY_RESTAURANT_RECEIVED]: (state, { historyId, restaurant }) => ({
+    ...state,
+    histories: update(
+      findIndex(propEq('id', historyId), state.histories), 
+      {
+        ...find(propEq('id', historyId), state.histories),
+        restaurant,
+      },
+      state.histories,
+    ),
```

Comment:
Are you sure this is correct?
If the `findIndex` doesn't fine the history id it will return `-1` which will update the last element in the `histories` array and I'm not sure this is what you want.